tests/e2e/vmgroupmisc_test.go

```diff
@@ -224,3 +225,37 @@ func (s *vgBasicSuite) TestDSAccessPrivilegeForUserVG(c *C) {
 
 	misc.LogTestEnd(c.TestName())
 }
+
+// Test steps:
+// 1. Create a vm VM1 using govc
+// 2. Create a vmgroup and associate VM1 to it.
+// 2. Execute vmgroup ls command.
+// 3. Delete the vm that was added to the vmgroup.
+// 4. Again execute vmgroup ls command to verify command works fine.
+func (s *vgBasicSuite) TestDeleteVMFromVmgroup(c *C) {
+	misc.LogTestStart(c.TestName())
+	vmName := "VM_" + inputparams.GetRandomNumber()
+	vgName := "VG_" + inputparams.GetRandomNumber()
+
+	// Create a vm - we need this to add to vmgroup and later on delete this vm
+	esx.CreateVM(vmName, s.config.Datastores[0])
+
+	admincli.CreateVMgroup(s.config.EsxHost, vgName, vmName, admincliconst.VMHomeDatastore)
+	// Verify if vmgroup exists
+	isVmgroupAvailable := admincli.IsVmgroupPresent(s.config.EsxHost, vgName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup ls command does not lists the vmgroup %s .", vgName))
+
+	// Verify vm belongs to vmgroup
+	isVMPartofVg := admincli.IsVMInVmgroup(s.config.EsxHost, vmName, vgName)
+	c.Assert(isVMPartofVg, Equals, true, Commentf("VM %s does not belong to vmgroup %s .", vmName, vgName))
+
+	// Destroy the vm
+	esx.DestroyVM(vmName)
+	misc.SleepForSec(3)
+
+	// Check vmgroup ls
+	isVmgroupAvailable = admincli.IsVmgroupPresent(s.config.EsxHost, vgName)
+	c.Assert(isVmgroupAvailable, Equals, true, Commentf("vmgroup ls command does not lists the vmgroup %s .", vgName))
+
```

Comment:
It is better to add TODO here saying .. due to product behavior vmgroup deletion is not possible and add the reference of the issue here .. if there is any issue otherwise please raise one and add its reference here.